chromadb/server/fastapi/__init__.py

```diff
@@ -340,8 +363,15 @@ def create_database(
             ),
         ),
     )
-    def get_database(self, database: str, tenant: str = DEFAULT_TENANT) -> Database:
-        return self._api.get_database(database, tenant)
+    async def get_database(
+        self, database: str, tenant: str = DEFAULT_TENANT
```

Comment:
I'm a little confused -- why doesn't this one also accept the raw request object? I would like us to standardize on either using FastAPI built-in request deserialization or manual deserialization.

I had this question a few times when reading the code, not going to repeat but instead just ask here

Reply:
The convention I followed is:

- Path, query params - leave them as params
- Body payloads - use request object to read the body asynchronously

I think even FastAPI does this distinction where anything it can get from the HTTP request headers, like query and path, will do so synchronously, whereas the body stream is read async.

We can, of course, fetch the path/query from the request object itself, but I thought leaving the path params makes it a little easier to reason about things, which is invalidated by your comments :)

Reply:
That makes sense to me! Thank you for explaining. I'm happy to do it either way.

Reply:
Let's leave it as is of now then.